sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2050 @@
+        // provenance
+        XmpMeta::register_namespace("http://purl.org/dc/terms/", "dcterms")
+            .map_err(xmp_write_err)?;
+        // Build a simple XMP meta element from the current source stream
+        let mut xmp_meta = match build_xmp_from_stream(source) {
+            // Use the data already available
+            Ok(meta) => meta,
+            // If data was not found for building out the XMP, we will default
+            // to some good starting points
+            Err(Error::NotFound) => default_font_xmp_meta(None, None)?,
+            // At this point, the font is considered to be invalid possibly
+            Err(error) => return Err(error),
+        };
+        // Reset the source stream to the beginning
+        source.seek(SeekFrom::Start(0))?;
+        // We don't really care if there was a provenance before, since we are
+        // writing a new one we will either be adding or overwriting what
+        // was there.
+        xmp_meta
+            .set_property(
+                "http://purl.org/dc/terms/",
+                "provenance",
+                &manifest_uri.into(),
+            )
+            .map_err(xmp_write_err)?;
+        // Finally write the XMP data as a string to the stream
+        add_reference_to_stream(source, destination, &xmp_meta.to_string())?;
+
+        Ok(())
+    }
+}
+
+struct TempFile {
+    // The temp_dir must be referenced during the duration of the use of the
+    // temporary file, as soon as it is dropped the temporary directory and the
+    // contents thereof are deleted
+    #[allow(dead_code)]
+    temp_dir: TempDir,
+    path: Box<Path>,
+    file: File,
+}
+
+impl TempFile {
+    /// Creates a new temporary file within the `env::temp_dir()` directory,
+    /// which should be deleted once the object is dropped.
+    ///
+    /// ## Arguments
+    ///
+    /// * `base_name` - Base name to use for the temporary file name
+    pub fn new(base_name: &Path) -> Result<Self> {
+        let temp_dir = TempDir::new()?;
+        let temp_dir_path = temp_dir.path();
+        let path = temp_dir_path.join(
+            base_name
+                .file_name()
+                .ok_or_else(|| Error::BadParam("Invalid file name".to_string()))?,
+        );
+        let file = File::create(&path)?;
+        Ok(Self {
+            temp_dir,
+            path: path.into(),
+            file,
+        })
+    }
+
+    /// Get the path of the temporary file
+    pub fn get_path(&self) -> &Path {
+        self.path.as_ref()
+    }
+
+    /// Get a mutable reference to the temporary file
+    pub fn get_mut_file(&mut self) -> &mut File {
+        &mut self.file
+    }
+}
+
+/// Supported extension and mime-types
+static SUPPORTED_TYPES: [&str; 4] = [
+    "application/font-woff",
+    "application/x-font-woff",
+    "font/woff",
+    "woff",
+];
+
+/// Four-character tag which names a font table.
+///
+/// TBD - Debug needs help,
```

Comment:
Can add `#[derive(Debug)]` for an automated debug formatting...

Reply:
Not good enough for VS Code, see my complaints in the Retro...

Reply:
I read the retro complaints, I still don't understand why we can use `#[derive(Debug)]`, maybe can explain more after daily?

Reply:
(The problem isn't that `Debug` doesn't work, the problem is that the VS Code debugger and/or `LLDB` and/or `vsdbg` don't use `Debug` or `Display` -- see [C2PA-271](https://monotype.atlassian.net/browse/C2PA-271).)

Reply:
I see, you just mean the "Debug" experience not the "Debug output".